app/abci.go

```diff
@@ -0,0 +1,160 @@
+package app
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"sort"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/lazyledger/lazyledger-app/x/lazyledgerapp/types"
+	abci "github.com/lazyledger/lazyledger-core/abci/types"
+	core "github.com/lazyledger/lazyledger-core/proto/tendermint/types"
+)
+
+// This file should contain all of the althered ABCI methods
+
+// PreprocessTxs fullfills the lazyledger-core version of the ACBI interface, by
+// performing basic validation for the incoming txs, and by cleanly separating
+// share messages from transactions
+// todo(evan): refactor out a for loop.
+func (app *App) PreprocessTxs(txs abci.RequestPreprocessTxs) abci.ResponsePreprocessTxs {
+	squareSize := app.SquareSize()
+	shareCounter := uint64(0)
+	var shareMsgs []*core.Message
+	var processedTxs [][]byte
+	for _, rawTx := range txs.Txs {
+		// decode the Tx
+		tx, err := app.txConfig.TxDecoder()(rawTx)
+		if err != nil {
+			continue
+		}
+
+		// don't do anything if the transaction doesn't contain a PayForMessage sdk.Msg
+		if !hasWirePayForMessage(tx) {
+			processedTxs = append(processedTxs, rawTx)
+			continue
+		}
+
+		// only support transactions that contain a single sdk.Msg for now
+		if len(tx.GetMsgs()) != 1 {
+			continue
+		}
+
+		msg := tx.GetMsgs()[0]
+
+		// run basic validation on the transaction
+		err = tx.ValidateBasic()
+		if err != nil {
+			continue
+		}
+
+		// process and validate the message
+		coreMsg, signedTx, err := app.processMsg(msg)
+		if err != nil {
+			continue
+		}
+
+		// execute the tx in runTxModeDeliver mode (3)
+		// execution includes all validation checks burning fees
+		// currently, no fees are burned
+		_, _, err = app.BaseApp.TxRunner()(3, rawTx)
```

Comment:
Extract 3 into a constant. Can't the mode be imported here instead?

Reply:
mode cannot be imported, as it is [not exported](https://github.com/lazyledger/cosmos-sdk/blob/3addd7f65d1c35f76716d67641ee629b01f5b9c7/baseapp/baseapp.go#L28) (the runTx method is not exported either, that's why we need the slightly modified sdk branch). I did move it to a constant and left a comment explaining its purpose and it is copy pasted